### PR TITLE
Menu option to allow selection of acs_type - adds #5371

### PR DESCRIPTION
**Pull request by ohnoitsavon**
**adds #5371**

This adds a menu which filters ribbon selection based on acs_type.

screenshot_acs_menu

**Modified functions:**

_monika_ribbon_select:_
- Added code to create a list of acs_types and pass this onto the sidebar select function.
- Moved Monika to the left to allow space for the menu.

_mas_selector_sidebar_select_acs:_
- Accepts a new variable acs_list, this holds the list of acs_types.
mas_selector_sidebar_select:
- Accepts a new variable acs_list, this holds the list of acs_types. 
- A new variable acs_type is created, this holds the currently selected acs_type.

_mas_selector_sidebar:_
- Accepts a new variable acs_list, this holds the list of acs_types.
- The sidebar now only shows items which match the acs_type. 
- If acs_type is None all items are shown.
- The new menu is drawn here displaying a button for each of the acs_types in acs_list.
- When a type is selected, acs_type is updated. The screen is updated refiltering the results.
- Restoring the selection resets the acs_type to Null.

**New functions:** 

_mas_acs_name_format:_
- Takes in a string intended to be in the current format of how the acs_types are stored.
- Removes “-“, adds spaces, adds a capital to the first letter, and pluralizes.
- Returns formatted string.

_mas_plularize:_
- A basic pluralizer used in mas_acs_name_format.
- Allows exceptions to be added along with general rules.

### CODE REVIEW

**Comment by ThePotatoGuy** [comment]
pushing this to next rel because of time.

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [inline Monika After Story/game/zz_selector.rpy:3746]: Looking at this, I think there might be a way we can do this where we can have more control over the button names and what elements are part of each category. 

If we use a dict with the key as the button name and the value as a list of selectables associated wtih that key, then in the screen we can check if a button is selected and only show the selectables for that key. (and if no buttons are selected, show the full item list). We can pass this dict into the selector labels.

For generic ACS type cases (like this case, for example), we can have a function in the `mas_selspr` store that does the mapping based on ACS type, and another function for adjusting the keys so they are appropriate for visuals.

This way would give us more flexibility in which buttons to show and which selectables to categorize under a button, (including the option of having a selectable be under multiple buttons), which will help with making this work with hair/clothes selectors.

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]

File: Monika After Story/game/definitions.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/definitions.rpy]: I have added a new commit that hopefully addresses this

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy]: I have added a new commit that hopefully addresses this

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy]: I have added a new commit that hopefully addresses this

**Comment by ohnoitsavon** [comment]
I have added a new commit that hopefully addresses the changes. I have also seen that it would be possible to add filter options for hair and clothes using ex_props, but that would rely on the addition of AOC and the different categories.

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [inline Monika After Story/game/zz_selector.rpy:1772]: two spaces between functions when in a module

```suggestion

    
    def mas_item_name_format(item_name):
```

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [inline Monika After Story/game/zz_selector.rpy:3475]: I would rather we not have global variables affecting the screen. If you need something to maintain state within screen iterations, then add additional props to the mailbox that is used here.

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [inline Monika After Story/game/zz_selector.rpy:3133]: midloop jumping is mainly for showing the dialogue box. The way screens work is that they refresh continuously, so setting the variable is enough to make the screen change next refresh.

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy]: Fixed in new commit.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy]: Fixed in new commit.

**Comment by ohnoitsavon** [comment]
I have added a new commit that addresses the changes.

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [inline Monika After Story/game/zz_selector.rpy:3250]: The `in` check here is expensive since its going thru a list.

Rather than this, if a filter is set, we should set a screen var with the selectables from the `filter_map`, then have the search code (`selector_search_callback`) adjusted so it checks that screen var first. 

However, this would also need to manually trigger search to re-run whenever the filter is selected since the search callback is only on input change. I'm not sure if we save the search_query anywhere (@Booplicate might know) but that would have to be saved somewhere (ideally mailbox) so we manually trigger search.

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]

File: Monika After Story/game/zz_selector.rpy
Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:3251]: I want to clarify, you suggest to append the selected filter option to the search query?

Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:3251]: The search callback indeed can access the screen scope, so we can set/get a screen var (or mailbox var). Although, there's a few things I want to mention.

If the user searched, we shouldn't change the query. That's counterintuitive, plus we might not be able to remove that from the query later (say if the user removes/changes the filter).

While the search doesn't seem to cause any slowdown, it is kinda heavy. It checks name, id, type, and ex-props of **every provided item** (consider also all the callbacks it has to execute). Its flexibility comes at the cost of speed, I don't think we should rerun it on filter change. As of right now filtering seems to be more specific, but faster.

Comment (ThePotatoGuy) [reply Monika After Story/game/zz_selector.rpy:3251]: ok going to have some definitions here:
* button filter - the filter buttons (aka the buttons that map to a list of selectables)
* search filter - the filtering that occurs when a search is run

>If the user searched, we shouldn't change the query. That's counterintuitive, plus we might not be able to remove that from the query later

I did not mean change the search query, I mean save the search query to retrigger the search on the button-filtered set of items. 

Right now the way this works is the search code filters down `items`, and then the loop that adds the selectables to the screen does another loop per item to check if its in the list of selectables from the button filter.

A better solution would be that the search (which is the most expensive) runs on the limited set of selectables given by the button-filter (which is a direct mapping) if possible. But the problem is that if you ran a search prevoiusly, then use the button filter, the search code isn't going to re-run since its a callback on the input field.

pasting from @ohnoitsavon 's DM, (btw just respond on here for documentation purposes):
>i have worked out exactly what you meant now in terms of "screen variable" and can see how it would work, however it is not possible as the search sub functions all revolve arround "MASSelectableImageButtonDisplayable" and the items are not that, they are just the normal selectables

Yup, not possible at least how I was originally thinking about it. But another option could be:  
* in the pre-screen label (`mas_selector_sidebar`), loop over `filter_map` and replace selectables with their disp equivalent in items - (sort of expensive, but its a run-once so cheaper in the long run). This will be what is passed into the actual screen.
* still have a screen var for the subset of selectable displayables - search code should check for this var first
* when the button filter is selected do 2 things:
    * set the screen var to the list of selectable disps based on `filter_map`
    * loop over `flt_items` and remove ones that don't belong - I think the loop is probably faster than re-running the search, but the  key point is that its only ran when a button filter is selected instead of constantly via screen refresh.

Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3251]: @Booplicate @ThePotatoGuy I think I can do something very close to that, I am looking into converting them in mas_selector_sidebar.

Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:3251]: > A better solution would be that the search (which is the most expensive) runs on the limited set of selectables given by the button-filter (which is a direct mapping) if possible. But the problem is that if you ran a search prevoiusly, then use the button filter, the search code isn't going to re-run since its a callback on the input field.

Ah, okay that makes perfect sense now. Yeah, just save the query in the mailbox and call the search callback again as a button action.

> I am looking into converting them in mas_selector_sidebar.

Making displayables in the screen isn't ideal since that code will be: 1. rerun on every interaction, 2. rerun as we loop.

Comment (ThePotatoGuy) [reply Monika After Story/game/zz_selector.rpy:3251]: updated my comment - I meant `mas_selector_sidebar_select`, which is the pre-loop label

Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3251]: New commit submitted

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [inline Monika After Story/game/zz_selector.rpy:3474]: this should use the same selectable displayables as the ones in `disp_items`. (So you should loop over the `disp_items` for each selectable in `filter_map` to get the displayables after `disp_items` is set) 

Otherwise there could be weirdness since this currently results in two displayables per selectable.

**Review by ThePotatoGuy — CHANGES_REQUESTED** [review]

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3474]: I have done a new commit that should fix this

**Comment by ohnoitsavon** [comment]
This hopefully addresses the issues raised.



I have a question regarding the use of this in other menus. The way the code is setup allows any method of assigninh categories as long as the items are added to the cateogories, this section of code currently does that for the ribbons:

`# generate filter map
        mapping = {}
        for acs_sel in use_acs:
            acs_type_tc = mas_selspr.mas_item_name_format(acs_sel.get_sprobj().acs_type)
            if acs_type_tc in mapping:
                mapping[acs_type_tc].append(acs_sel)
            else:
                mapping[acs_type_tc] = [acs_sel]`
                
If elements are added for clothes/hair categories for example with the addition of AOC, these could be used to create the categories, or they could be made by other methods by checking for specific exprops. Are there any plans for this, or any suggestion of how this should be implemented?

**Comment by ThePotatoGuy** [comment]
>Are there any plans for this, or any suggestion of how this should be implemented?

I'm not sure what you are asking here. The changes I asked you to make before were to make it so we can pass in a data structure like `key: category title, value: list of selectables for the category`. This alone means that if we wanted the categories to be exprops or anything else, it would be just a matter of sorting the data into this structure (aka, thats how future filter categories would be implemented).

**Review by ThePotatoGuy — APPROVED** [review]

**Comment by ohnoitsavon** [comment]
@ThePotatoGuy Your last comment explains it, I was just asking if there are any plans soon to do that, you said "future filter categories" It is something I _might_ want to look at, once this has been fully tested/approved.

File: Monika After Story/game/zz_selector.rpy
Comment (Booplicate) [inline Monika After Story/game/zz_selector.rpy:1787]: I don't like that we will have to maintain this list in the future. Plus custom user acs types won't be in this list anyway. Maybe we should just show types as they are?

**Review by Booplicate — CHANGES_REQUESTED** [review]
I noticed that the screen flicks when I change the filter. Is it possible to fix that? It doesn't happen when I interact in any other way (search bar, selecting an acs, restoring state, etc).

I'd also say we should support selecting multiple acs types at once, not just one.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:1787]: Showing "-"'s in names would be very untidy, and removing it would make it "s type ribbon" which is wrong, @ThePotatoGuy can you make a decision on this? I think the better option is to add a new exprop for cateogory, anything without it would be uncatorgorised, also the name would be what is shown in the menu then.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3479]: Sorry for the noob question but what do you mean?

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3201]: What do you suggest instead?

**Comment by ohnoitsavon** [comment]
> 
> 
> I noticed that the screen flicks when I change the filter. Is it possible to fix that? It doesn't happen when I interact in any other way (search bar, selecting an acs, restoring state, etc).
> 
> I'd also say we should support selecting multiple acs types at once, not just one.

Mine does not flick, also I would not know how to fix that. as I am just doing the same as when you select an outift to wear/when you search. Do you have any suggestion by looking at the code as to what is different?

I also do not know how it would be possible to add multiple options as I have not done that before in renpy, if you have any help or suggestion as to how that can be done let me know.

File: Monika After Story/game/zz_selector.rpy
Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:3201]: Rename those vars so anyone who sees this code for the first time can distinguish that one of them is for search bar, another is for filter menu.

File: Monika After Story/game/zz_selector.rpy
Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:3479]: Right now `filter_map_selbtn` is a `dict`. `defaultdict` accepts a factory as its argument which allows automatically fill missing keys.
E.g. `filter_map_selbtn = collections.defaultdict(list)`. When you access a key that's missing from `filter_map_selbtn`, it will make an empty `list` for that key. So you can just do `filter_map_selbtn["this_key_doesnt_exist"].append(item)` w/o doing that `in` check. [More examples](https://docs.python.org/3/library/collections.html#defaultdict-examples).

File: Monika After Story/game/zz_selector.rpy
Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:1787]: I think both of those variants are better than having to remember that we should maintain a map that's local to a function scope.

Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:1787]: I will remove the exception list then, no worries for me, but i feel that somone needs to look at adding new property which could apply to clothes and hair too, as this code can work for that, but I feel that is a developer decision, as it would alter all the outifts but would allow beter sections and would be needed for adding support for @multimokia 's AOC/AHC

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3201]: Those examples are what I was looking for thanks.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3479]: Thank you I undertsand what you mean now and will implement that

**Comment by Booplicate** [comment]
> I also do not know how it would be possible to add multiple options as I have not done that before in renpy, if you have any help or suggestion as to how that can be done let me know.

Some time ago I made [`mas_check_scrollable_menu`](https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/screens.rpy#L2816), which has similar functional - selecting multiple buttons. Maybe it can help. I'd think that instead of filtering by just one type, we'd have a `set` of types we want to show.

**Comment by ohnoitsavon** [comment]
> 
> 
> > I also do not know how it would be possible to add multiple options as I have not done that before in renpy, if you have any help or suggestion as to how that can be done let me know.
> 
> Some time ago I made [`mas_check_scrollable_menu`](https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/screens.rpy#L2816), which has similar functional - selecting multiple buttons. Maybe it can help. I'd think that instead of filtering by just one type, we'd have a `set` of types we want to show. Selecting a new filter in your menu, adds that type to the set, unselecting removes.

I will look at that, it should be easy then, but what about the flicker? I am not sure how that can be resolved, it is only for me when i dont have anything in the search, do you have any suggestion?

**Comment by ThePotatoGuy** [comment]
I disagree on being able to select multiple at a time. I think the common use cases for these filters would be:
1. finding a specific ACS by narrowing down by category (but being too lazy to type to search)
2. wanting to see all the options for a specific category

Single select covers these use cases easily.

Multiple select seems more suited to showing combinations of things, but since we use the filters to separate visually different ACS, a sprite that is a combination of these ACS categories wouldn't be a thing.
Multiple select also means flipping through categories would require an extra click since u need deselect the previous to limit to the new category.

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [reply Monika After Story/game/zz_selector.rpy:1787]: We should keep the exception list. We can add API functions to add mapping exceptions to the list, so user-defined ones can add code to add their name exceptions.

Exprop isn't that great for this since it means every spritepack with that `acs_type` must include an additional element that is basically a copy of the `acs_type`.

**Comment by multimokia** [comment]
>     2\. wanting to see all the options for a specific category

I'd actually argue that personally.

Using the options as categorical filters actually makes a lot of sense and is intuitive.

If you want to just refine down to xyz types from a larger pool, I think that helps a lot. Say if you wanted to exclude specifically s-types but see the rest to pick something. You've refined a much larger pool into a smaller one.

Not to mention, if you wanted to target one type specifically, it makes more sense to use search for that.

Additionally, search has no `not` filter, so regardless, if the amount of ribbons and other hair tie types continues to grow, the overarching problem still exists as even specific subcategories become large. As such being able to exclude them entirely makes more sense to mitigate it long-term.

**Comment by ThePotatoGuy** [comment]
Single select already refines the larger pool into a smaller one so I don't see that as a talking point for multiple select. Same goes for "search has no not filter", single select is literally the most exclusive filter you can get without typing.

I think the best example here is the Sims 4 character creation screen. When you select a clothing item, you can pick subcategories, but they aren't combinable when clicking on them. They are separate and do a great job of narrowing the available options. If you want to see something from another category then you can switch to that category.

**Comment by ohnoitsavon** [comment]
Well, I will make the code changes to fix the issues and look at any possible fix for the flicker, although but I do not know if it is possible. 

As for the categories why don't we have it as single for now and look at adding multiple if it is requested or shown as an issue? It also means this thing can adding and expansion into things like clothes and hair could be done? It would be amazing to have "short" "short tied" "long/down" "long tied" or something @ThePotatoGuy I assume that would be another or but is the same code I just don't know the categories / exprop if there is one that would show those values.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:1787]: Ok I will keep the exception list

**Comment by Booplicate** [comment]
> Multiple select seems more suited to showing combinations of things, but since we use the filters to separate visually different ACS, a sprite that is a combination of these ACS categories wouldn't be a thing.

Not sure, it feels kinda limited. Selecting multiple cats was the first thing I tried as a user, because it felt intuitive. Plus, I think it can be even more useful for clothes/hair selection in the future.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:1787]: @ThePotatoGuy @Booplicate I can remove it or keep it, there is no issue either way, but it is only if a - is in the name, which is a very small amount of things. I don't know how it would be made into an API or moved scopes, please advise on this, but for now, I will leave it in and make the other changes, the dev team can then decides on that privately and come back to me with a decision on it.

**Comment by ohnoitsavon** [comment]
> 
> 
> > Multiple select seems more suited to showing combinations of things, but since we use the filters to separate visually different ACS, a sprite that is a combination of these ACS categories wouldn't be a thing.
> 
> Not sure, it feels kinda limited. Selecting multiple cats was the first thing I tried as a user, because it felt intuitive. Plus, I think it can be even more useful for clothes/hair selection in the future.

For now I will leave it and make the other changes, I feel this needs to be a dev team decision, as I am ok with either and have a good idea how to implement it IF it is required, but personally am siding with @ThePotatoGuy at least for ACS's.

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [reply Monika After Story/game/zz_selector.rpy:1787]: @Booplicate 
We have maps for a ton of other things in the mod, another mapping is hardly that much more complicated.
However, I think we can actually do an exprop route if we added it to the ACSTemplate for s-type-ribbons. 
What do you think about that?

**Comment by ThePotatoGuy** [comment]
@Booplicate @multimokia 
Even for clothes/hair filters (as exprops), I still think single select is more intuitive. Again, the sims 4 example can have clothes in multiple categories, but they still keep the subcats as single select.

Multiple select works for things like shopping for a product that has a ton of variations (for example, finding all computers that are under a certain amount, use a certain process, have a certain RAM size, etc...). With clothes/hair/ACS, however, we have a larger variety of products with little variation between the same products (hoodies vs bikins vs shirts, twin tails vs pony tails vs no tails, bow vs ribbon vs s-type ribbons (color is the main variation for ACS), etc...).

File: Monika After Story/game/zz_selector.rpy
Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:1787]: I think that could work, but we don't have a template for clothes/hair.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3201]: This works, tested in my code, will post in the next commit.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3479]: This works, tested in my code, will post in the next commit.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:4049]: This works, tested in my code, will post in the next commit.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3176]: This works, tested in my code, will post in the next commit.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3130]: This works, tested in my code, will post in the next commit.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3165]: This will be added to the commit.

File: Monika After Story/game/zz_selector.rpy
Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:3130]: This works and will be added to the commit

**Comment by ohnoitsavon** [comment]
I have addressed all changes in the latest commit including fixing the flashing screen following TPG's suggestions. I have left the name space for s-type code in for now, but await further guidance on that.

@ThePotatoGuy / @Booplicate What needs to be done for that?

File: Monika After Story/game/zz_selector.rpy
Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:1787]: This hasn't been addressed yet.

Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:1787]: @Booplicate I was told by @ThePotatoGuy to leave it

Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:1787]: @Booplicate @ThePotatoGuy has the final say yes? so I would like clear action from them about what to do next

Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:1787]: I have been waiting almost 30 days for any kind of response to this, while I know this is not that important and people have lives to live, I feel we need an answer on this so we know what to do, ALL other chnages have been actioned.

Comment (Booplicate) [reply Monika After Story/game/zz_selector.rpy:1787]: I'm waiting for TPG response. But even if we kept the mapping, it shouldn't be local anyway.

Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy:1787]: Ok, one option would be if we add something like a "cateogory name" exprop, that could be added TO ANYTHING including hair clothes etc. Meaning the search has another thing it can check, and the menus can work with anything?

**Comment by Booplicate** [comment]
@ohnoitsavon I wrapped some lines to make it a bit more easier to read.

Also during testing I noticed that the `Filter` button and the current filter reset when you press the `Restore` button. Is that indented behavior? Because I think it's rather inconvenient.

**Comment by ohnoitsavon** [comment]
> 
> 
> @ohnoitsavon I wrapped some lines to make it a bit more easier to read.
> 
> Also during testing I noticed that the `Filter` button and the current filter reset when you press the `Restore` button. Is that indented behavior? Because I think it's rather inconvenient.

Hmm... the wrap is obviously fine I am checking the game and the filter doesnt reset but the menu does close when restore is clicked, is that what you mean?

**Comment by ohnoitsavon** [comment]
I have also noticed some other bugs when restoring the search gains an extra space? I will look into these changes, but feel there are some fixes need.

**Comment by ohnoitsavon** [comment]
Fixed the issues mentioned above with the menu and search

File: Monika After Story/game/zz_selector.rpy
Comment (ThePotatoGuy) [reply Monika After Story/game/zz_selector.rpy:1787]: keep the mapping, move it so it's in the `mas_selspr` name space and we'll make adjustments in the future as necessary.

Comment (ohnoitsavon) [reply Monika After Story/game/zz_selector.rpy]: I have added this in the new commit

**Comment by ohnoitsavon** [comment]
I have fixed the issue with the mapping moving it to the new location @ThePotatoGuy

**Review by ThePotatoGuy — APPROVED** [review]

**Comment by ohnoitsavon** [comment]
It says I have a requested change @Booplicate - what do I need to do? I though it was all sorted?

**Comment by Booplicate** [comment]
I haven't requested anything. We've reviewed your code, now we're testing your additions.

**Review by ThePotatoGuy — APPROVED** [review]

**Comment by ThePotatoGuy** [comment]
@Legendkiller21 has to test it so we have to wait for them.

**Review by Legendkiller21 — APPROVED** [review]

**Pull request merged**
